test_conformance/printf/util_printf.cpp

```diff
@@ -955,48 +1038,58 @@ size_t verifyOutputBuffer(char *analysisBuffer,testCase* pTestCase,size_t testId
             ++eCorrectBuffer;
         return strcmp(eCorrectBuffer,exp);
     }
-    if(!strcmp(pTestCase->_correctBuffer[testId].c_str(),"inf"))
-    return strcmp(analysisBuffer,"inf")&&strcmp(analysisBuffer,"infinity")&&strcmp(analysisBuffer,"1.#INF00")&&strcmp(analysisBuffer,"Inf");
-    if(!strcmp(pTestCase->_correctBuffer[testId].c_str(),"nan") || !strcmp(pTestCase->_correctBuffer[testId].c_str(),"-nan")) {
-       return strcmp(analysisBuffer,"nan")&&strcmp(analysisBuffer,"-nan")&&strcmp(analysisBuffer,"1.#IND00")&&strcmp(analysisBuffer,"-1.#IND00")&&strcmp(analysisBuffer,"NaN")&&strcmp(analysisBuffer,"nan(ind)")&&strcmp(analysisBuffer,"nan(snan)")&&strcmp(analysisBuffer,"-nan(ind)");
-    }
-    return strcmp(analysisBuffer,pTestCase->_correctBuffer[testId].c_str());
+
+    if (!strcmp(pTestCase->_correctBuffer[testId].c_str(), "inf"))
```

Comment:
Nice, that's much more readable now!  Optionally, you could also get rid of the `c_str` here and just use `==` string comparisons instead of `!strcmp`.

Reply:
Absolutely, done!